### DISCUSSION TITLE
Designing UI

### DISCUSSION THREAD

**Discussion by yinanazhou** [DDMAL/VIM#26] [General]
Hi, I'd like to discuss a few things before designing the UI for VIM🚀 
@fujinaga @dchiller @PythonSemicolon @Monalisa98927 , since I'm new to the group, thanks in advance for helping me fill in the gaps. 

### 1. Project Objectives & Goals:

One of the main goals of the project is to make Wikidata's instrument data more complete and more multi-lingual. So VIM is in some sense a "front-end" for adding this data to Wikidata.

### 2. Target Audience:
- Musicologists
- People with no musicological training should be able to add images and instrument names.

### 3. Content & Data:
- Model reference: https://github.com/DDMAL/VIM/discussions/7#discussion-5415048
- Search page design: https://github.com/DDMAL/VIM/issues/1#issuecomment-1645996407

### 4. Database Structure
- Model reference: https://github.com/DDMAL/VIM/discussions/7#discussion-5415048
- We are using a postgres database

### 5. Backend Functionality
- Backend will return appropriate data at the endpoints outlined in https://github.com/DDMAL/VIM/discussions/7
- Backend will also take care of all transactions with wikidata

### 7. User Stories & Use Cases
What are the main tasks users will perform?

- searching photos to see if an instrument exists on VIM
- tagging an existing instrument with a label in a specific language
- adding a new instrument if it does not exist on VIM
- uploading images, audio, video of instruments

To be discussed: https://github.com/orgs/DDMAL/projects/28/views/1

### 8. Authentication & User Management
- User enrollment and authentication is required for adding to or editing VIM
- Some individual user data should be visible only to that specific authenticated user
- Some users will have superuser/administrative functions.

### 9. Responsiveness & Cross-Browser Compatibility
N/A

### 10. Timeline & Milestones
TBD

**Comment by dchiller** [comment]
### 1. Project Objectives & Goals:

What you put is right...the only thing I'd emphasize is that one of the main goals of the project is to make Wikidata's instrument data more complete and more multi-lingual. So VIM is in some sense a "front-end" for adding this data to Wikidata.

### 2. Target Audience:

I suspect most people who interact with the site will have some musical background, but we want people who have no musicological training to be able to add images and instrument names.

### 3. Content & Data:

We've had a lot of discussion about database and site architecture and contents, so #17 feels a bit out of date. I would look at the primary post of #7 (https://github.com/DDMAL/VIM/discussions/7#discussion-5415048) which we've updated to reflect the results of those discussions.

### 4. Database Structure

I think the primary post of #7 is the best place to see the current thinking on database structure. We are using a postgres database.

### 5. Backend Functionality

At least initially, we'll focus I think on having the backend return appropriate data at the endpoints outlined in #7. The backend will also take care of all transactions with wikidata.

### 7. User Stories & Use Cases

The main tasks that the user will perform is:
- searching photos to see if an instrument exists on VIM
- tagging an existing instrument with a label in a specific language
- adding a new instrument if it does not exist on VIM
- uploading images, audio, video of instruments

Let me take a stab at some stories and then others can comment/narrow down for what we start with.

### 8. Authentication & User Management

User enrollment and authentication is required for adding to or editing VIM. Some individual user data should be visible only to that specific authenticated user. Some users will have superuser/administrative functions.

### 9. Responsiveness & Cross-Browser Compatibility

We've not discussed, so this is just my take: not really expecting anyone to use this on a mobile device.

### 10. Timeline & Milestones

I think we need to get more clear with this. The big milestone right now is the LinkedMusic project meeting in October, but what exactly we are trying to get done for that is not exactly clear. Let's get some user stories decided and see what makes sense to target for October.

**Comment by dchiller** [comment]
As an update to the above, I drafted a few user stories for the Oct meeting at https://github.com/orgs/DDMAL/projects/28. Perhaps we can confirm at the coming meeting whether these seem like a good group to shoot for or we should do more/fewer...

**Reply by yinanazhou** [reply]
This is super! Thank you!

**Comment by yinanazhou** [comment]
The color palette used in Neon:

```
/**
 * Adapted from color palette from Figure 2 (Colors optimized for color-blind
 * individuals) from
 * ["Points of view: Color blindness" by Bang Wong published in Nature Methods volume 8 on 27 May 2011](https://www.nature.com/articles/nmeth.1618?WT.ec_id=NMETH-201106)
 */
```
